File: RayTracerCSSingleProj/SurfaceTexture.cs
using System;

namespace RayTracer
{
    public class SurfaceTexture
    {
        public Func<Vector3, Colour> Diffuse { get; private set; }
        public Func<Vector3, Colour> Specular { get; private set; }
        public Func<Vector3, double> Reflect { get; private set; }
        public double Roughness { get; private set; }

        public SurfaceTexture(Func<Vector3, Colour> diffuse, 
            Func<Vector3, Colour> specular,
            Func<Vector3, double> reflect,
            double roughness)
        {
            Diffuse = diffuse;
            Specular = specular;
            Reflect = reflect;
            Roughness = roughness;
        }
    }
}

File: RayTracerVBSingleProj/SurfaceTexture.vb
Public Class SurfaceTexture

    Public ReadOnly Property Diffuse As Func(Of Vector3, Colour)

    Public ReadOnly Property Specular As Func(Of Vector3, Colour)

    Public ReadOnly Property Reflect As Func(Of Vector3, Double)

    Public ReadOnly Property Roughness As Double

    Public Sub New(ByVal diffuse As Func(Of Vector3, Colour), ByVal specular As Func(Of Vector3, Colour), ByVal reflect As Func(Of Vector3, Double), ByVal roughness As Double)
        Me.Diffuse = diffuse
        Me.Specular = specular
        Me.Reflect = reflect
        Me.Roughness = roughness
    End Sub
End Class
